v0/destinations/pipedrive/util.js

```diff
@@ -0,0 +1,247 @@
+    return null;
+  }
+};
+
+const createNewOrganisation = async (data, destination) => {
+  const resp = await axios
+    .post(ORGANISATION_ENDPOINT, data, {
+      params: {
+        api_token: destination.Config.api_token
+      },
+      headers: {
+        "Content-Type": "application/json"
+      }
+    })
+    .catch(err => {
+      throw new Error(`failed to create new organisation: ${err}`);
+    });
+
+  if (resp && resp.status === 201) return resp.data.data;
+  throw new Error("failed to create new organisation");
+};
+
+/**
+ * Updates already existing organisation with
+ * the supplied payload and groupId
+ * @param {*} groupId
+ * @param {*} groupPayload
+ * @param {*} destination
+ */
+const updateOrganisationTraits = async (groupId, groupPayload, destination) => {
+  try {
+    const response = await axios.put(
+      `${ORGANISATION_ENDPOINT}/${groupId}`,
+      groupPayload,
+      {
+        params: {
+          api_token: destination.Config.api_token
+        },
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/x-www-form-urlencoded"
+        }
+      }
+    );
+
+    if (!response || response.status !== 200)
+      throw new Error("error while updating group");
+  } catch (err) {
+    throw new Error(`error while updating group: ${err}`);
+  }
+};
+
+const mergeTwoPersons = async (previousId, userId, destination) => {
+  const payload = {
+    merge_with_id: userId
+  };
+  try {
+    const mergeResponse = await axios.put(
+      getMergeEndpoint(previousId),
+      payload,
+      {
+        params: {
+          api_token: destination.Config.api_token
+        },
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/x-www-form-urlencoded"
+        }
+      }
+    );
+    if (!mergeResponse || mergeResponse.status !== 200)
+      throw new Error("merge failed");
+  } catch (err) {
+    throw new Error(`error while merging persons: ${err}`);
+  }
+};
+
+/**
+ * Wrapper on top of getFieldValueFromMessage
+ * If value is not found, throws custom error
+ * @param {*} message
+ * @param {*} field
+ * @param {*} err
+ * @returns
+ */
+const getFieldValueOrThrowError = (message, field, err) => {
+  const val = getFieldValueFromMessage(message, field);
+  if (!val) throw err;
+  return val;
+};
+
+/**
+ * * Util function to rename the custom fields
+ * based on fieldsMap in destination Config
+ * Supported type: product, track, leads, organization, person
+ * @param {*} message 
+ * @param {*} fieldsMap 
+ * @param {*} type 
+ * @returns 
+ */
+ const renameCustomFields = (message, fieldsMap, type) => {
+  const specificMap = fieldsMap[type];
+  if(!specificMap)  throw new Error(`fieldsMap doest not contain type ${type}`);
+
+  const payload = {};
+  Object.keys(message).map(key => {
+    if(Object.keys(specificMap).includes(key))
+      set(payload, specificMap[key], message[key]);
+    else
+      set(payload, key, message[key]);
```

Comment:
why this? then the property will be sent with the original name and it will be rejected right? we should delete the keys for which the mapping of key to API key is not there

Reply:
Also if name,org_id etc is mapped by user in webapp those will also be converted to API key. shouldn't we handle that? loop through the keys except the pipedrive defined keys.

Reply:
okay. i thought of letting that pass by and destination would reject it. then should we delete it with a warning log or something ?

Reply:
Nono it will always be rejected then. so we should drop. yes with a warning.